src/spock_recovery.c

```diff
@@ -765,3 +766,187 @@ spock_find_rescue_source_sql(PG_FUNCTION_ARGS)
 
 	PG_RETURN_DATUM(HeapTupleGetDatum(tuple));
 }
+
+/*
+ * spock_clone_recovery_slot_sql
+ *
+ * Clone an existing inactive Recovery Slot into a new, temporary active slot
+ * for disaster recovery operations. The cloned slot can be used for selective
+ * replay of unacknowledged transactions.
+ *
+ * Usage: SELECT * FROM spock.clone_recovery_slot('origin_node', 'target_node');
+ *
+ * Returns a record with:
+ * - cloned_slot_name: Name of the newly created active slot
+ * - original_slot_name: Name of the original recovery slot
+ * - restart_lsn: LSN position to start replay from
+ * - success: Whether the cloning was successful
+ * - message: Status message about the operation
+ */
+Datum
+spock_clone_recovery_slot_sql(PG_FUNCTION_ARGS)
+{
+	text	   *origin_node_text = PG_GETARG_TEXT_PP(0);
+	text	   *target_node_text = PG_GETARG_TEXT_PP(1);
```

Comment:
This looks like Its trying to clone a slot from a origin node to a local node. Why do we need that?

Correct me If I'm wrong, IIRC, the idea was to clone the existing recovery slot, on the same node thats selected as source. That was to prevent the original recovery slot to change, until actual recovery is done for safety or restart purposes etc.